src/io/flutter/view/FlutterViewFactory.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.view;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.project.DumbService;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.wm.ToolWindow;
+import com.intellij.openapi.wm.ToolWindowFactory;
+import com.intellij.openapi.wm.ToolWindowManager;
+import org.jetbrains.annotations.NotNull;
+
+public class FlutterViewFactory implements ToolWindowFactory {
+  public static void init(@NotNull Project project) {
+    //noinspection CodeBlock2Expr
+    project.getMessageBus().connect().subscribe(
+      FlutterViewMessages.FLUTTER_DEBUG_TOPIC, (event) -> {
+        openFlutterView(project, event);
+      });
+  }
+
+  private static void openFlutterView(Project project, FlutterViewMessages.FlutterDebugEvent event) {
```

Comment:
I'd annotate `project` as `@NotNull` since that's what `ToolWindowManager.getInstance(project)` requires.